bigchaindb/events.py

```diff
@@ -1,33 +1,91 @@
-from enum import Enum
+from collections import defaultdict
 from multiprocessing import Queue
 
 
-class EventTypes(Enum):
+POISON_PILL = 'POISON_PILL'
+
+
+class EventTypes:
+    """Container class that holds all the possible
+    events BigchainDB manages."""
+
+    # If you add a new Event Type, make sure to add it
+    # to the docs in docs/server/source/event-plugin-api.rst
+    ALL = ~0
```

Comment:
It is used [here](https://github.com/bigchaindb/bigchaindb/pull/1707/files#diff-60d798ce60a59a778f9a841f17bef565R79) to dispatch an event to the correct queues.

An event it is a number that is a power of two. If you represent this number you have just a `1` followed by zero or more `0`s.

A subscriber can listen to a combination of events just by `or`ing the ids of those events, the name for this is *bitmask*.

When the dispatcher gets a new event, it `and`s the event's id with all the available *bitmasks*. If the result is greater than `0`, then the event belongs to that bitmask, and the dispatcher forwards the event to the subscribers.